ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -995,4 +1003,196 @@ void markStartTime() {
+  }
+
+  /**
+   * Injects {@link Account} and {@link Container} for the PUT requests that carry the target account and container headers.
+   * @param restRequest The {@link RestRequest} to inject {@link Account} and {@link Container} object.
+   * @throws RestServiceException if either of {@link Account} or {@link Container} object could not be identified.
+   */
+  private void injectAccountAndContainerUsingAccountAndContainerHeaders(RestRequest restRequest)
+      throws RestServiceException {
+    String accountName = getTargetAccountNameFromHeader(restRequest);
+    Account targetAccount = accountService.getAccountByName(accountName);
+    if (targetAccount == null) {
+      frontendMetrics.unrecognizedAccountNameCount.inc();
+      throw new RestServiceException("Account cannot be found for accountName=" + accountName
+          + " in put request with account and container headers.", RestServiceErrorCode.InvalidAccount);
+    }
+    ensureAccountNameMatch(targetAccount, restRequest);
+    String containerName = getTargetContainerNameFromHeader(restRequest);
+    Container targetContainer = targetAccount.getContainerByName(containerName);
+    if (targetContainer == null) {
+      frontendMetrics.unrecognizedContainerNameCount.inc();
+      throw new RestServiceException(
+          "Container cannot be identified for accountName=" + accountName + " and containerName=" + containerName
+              + " in put request with account and container headers.", RestServiceErrorCode.InvalidContainer);
+    }
+    setTargetAccountAndContainerInRestRequest(restRequest, targetAccount, targetContainer);
+  }
+
+  /**
+   * Obtains the target {@link Account} and {@link Container} id from the blobId string, queries the {@link AccountService}
+   * to get the corresponding {@link Account} and {@link Container}, and injects the target {@link Account} and
+   * {@link Container} into the {@link RestRequest}.
+   * @param blobIdStr The blobId string to get the target {@link Account} and {@link Container} id。
+   * @param restRequest The rest request to insert the target {@link Account} and {@link Container}.
+   * @throws RestServiceException if 1) either {@link Account} or {@link Container} could not be identified; or 2)
+   *                              either {@link Account} or {@link Container} were explicitly specified as
+   *                              {@link Account#UNKNOWN_ACCOUNT} or {@link Container#UNKNOWN_CONTAINER}.
+   */
+  private void injectTargetAccountAndContainerFromBlobId(String blobIdStr, RestRequest restRequest)
+      throws RestServiceException {
+    try {
+      BlobId blobId = new BlobId(blobIdStr, clusterMap);
+      Account targetAccount = accountService.getAccountById(blobId.getAccountId());
+      if (targetAccount == null) {
+        frontendMetrics.getHeadDeleteUnrecognizedAccountCount.inc();
+        // @todo The check can be removed once HelixAccountService is running with UNKNOWN_ACCOUNT created.
```

Comment:
We can, but not for every ambry user who change to containerization. For now I prefer to keep this code to allow the situation that Ambry starts before `UNKNOWN_ACCOUNT` is added into AccountService.